### ISSUE TITLE
[java] JUnitAssertionsShouldIncludeMessage reports false postive on AssertJ assertions

### ISSUE THREAD

**Issue by ajeans** [pmd/pmd#3350]
**Affects PMD Version:** master and 6.34

**Rule:** JUnitAssertionsShouldIncludeMessage

https://pmd.sourceforge.io/pmd-6.35.0/pmd_rules_java_bestpractices.html#junitassertionsshouldincludemessage

**Description:**

JUnitAssertionsShouldIncludeMessage reports a false positive on one of our test assertions (AssertJ `assertThat`)

**Code Sample demonstrating the issue:**

```java
import static org.assertj.core.api.Assertions.assertThat;

import java.util.Map;
import org.junit.jupiter.api.Test;

class SimpleTest {
  @Test
  public void simpleMethod() {
    assertThat(Map.of("foo", "FOO"))
        .containsEntry("foo", "FOO")
        .containsOnlyKeys("foo");;
  }
}
```

**Expected outcome:**

No error on the `assertThat` line
 

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Running with gradle and inside the PMD project with the attached patch to demonstrate the problem.

```diff
diff --git a/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/JUnitAssertionsShouldIncludeMessage.xml b/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/JUnitAssertionsShouldIncludeMessage.xml
index 0bb995b977..c3b69a74f3 100644
--- a/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/JUnitAssertionsShouldIncludeMessage.xml
+++ b/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/JUnitAssertionsShouldIncludeMessage.xml
@@ -430,6 +430,24 @@ class SimpleTest {
     public void simpleMethod() {
         assertEquals(0, Integer.compare(0, 0), "Not equals 0 != 1");
     }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>[java] JUnitAssertionsShouldIncludeMessage - False positive with AssertJ assertThat and JUnit5</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class SimpleTest {
+  @Test
+  public void simpleMethod() {
+    assertThat(Map.of("foo", "FOO"))
+        .containsEntry("foo", "FOO")
+        .containsOnlyKeys("foo");;
+  }
 }
         ]]></code>
     </test-code>
```

[false-positive.txt](https://github.com/pmd/pmd/files/6692393/false-positive.txt)

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
